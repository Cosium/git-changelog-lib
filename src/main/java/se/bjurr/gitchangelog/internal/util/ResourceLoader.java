package se.bjurr.gitchangelog.internal.util;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.stream.Collectors;

public final class ResourceLoader {
  private ResourceLoader() {}

  public static String getResourceOrFile(final String resourceName) {
    String templateString = null;
    try {
      final Path templatePath = Paths.get(resourceName);
      if (templatePath.toFile().exists()) {
        templateString = new String(Files.readAllBytes(templatePath), StandardCharsets.UTF_8);
      } else {
        InputStream inputStream =
            getResourceFromClassLoader(resourceName, ResourceLoader.class.getClassLoader());
        if (inputStream == null) {
          inputStream =
              getResourceFromClassLoader(
                  resourceName, Thread.currentThread().getContextClassLoader());
        }

        if (inputStream == null) {
          throw new FileNotFoundException(
              "Was unable to find file, or resouce, \"" + resourceName + "\"");
        }
        try (BufferedReader br =
            new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
          templateString = br.lines().collect(Collectors.joining("\n"));
        }
      }
    } catch (final IOException e) {
      throw new RuntimeException(resourceName, e);
    }
    return templateString;
  }

  private static InputStream getResourceFromClassLoader(
      final String resourceName, final ClassLoader classLoader) {
    InputStream inputStream = classLoader.getResourceAsStream(resourceName);
    if (inputStream == null) {
      inputStream = classLoader.getResourceAsStream("/" + resourceName);
    }
    return inputStream;
  }
}
