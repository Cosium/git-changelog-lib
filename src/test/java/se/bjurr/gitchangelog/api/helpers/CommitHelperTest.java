package se.bjurr.gitchangelog.api.helpers;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import com.github.jknack.handlebars.Helper;
import com.github.jknack.handlebars.Options;
import java.io.IOException;
import java.util.Arrays;
import java.util.List;
import java.util.Map.Entry;
import org.approvaltests.Approvals;
import org.junit.Test;
import se.bjurr.gitchangelog.test.AutoApproveReporter;

public class CommitHelperTest {

  private final List<String> commits =
      Arrays.asList( //
          "feat: subject", //
          "chore(dependencies): subject", //
          "chore(dependencies:formatting): subject", //
          "chore: subject (refs ABC-123)", //
          "chore: subject (refs ABC-123 DEF-456)", //
          "chore: subject (fixes ABC-123" //
          );

  @Test
  public void testCommit() throws IOException {

    final StringBuilder sb = new StringBuilder();

    for (final String commit : this.commits) {
      sb.append("----------------------------\n");
      sb.append(commit + "\n\n");
      for (final Entry<String, Helper<?>> helper : Helpers.COMMIT_HELPERS.entrySet()) {
        final Options options = mock(Options.class);
        when(options.hash("type")).thenReturn("feat");
        final Helper<String> value = (Helper<String>) helper.getValue();
        final String actual = value.apply(commit, options).toString();
        sb.append(helper.getKey() + ": " + actual + "\n\n");
      }
    }

    Approvals.verify(
        sb.toString(),
        new org.approvaltests.core.Options().withReporter(new AutoApproveReporter()));
  }
}
