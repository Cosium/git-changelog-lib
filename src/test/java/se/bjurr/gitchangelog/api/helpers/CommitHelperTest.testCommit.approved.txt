----------------------------
feat: subject

commitDescription: subject

commitFixes: []

commitRefs: []

commitScopes: []

commitType: feat

isType: true

revertedCommit: 

----------------------------
chore(dependencies): subject

commitDescription: subject

commitFixes: []

commitRefs: []

commitScopes: [dependencies]

commitType: chore

isType: false

revertedCommit: 

----------------------------
chore(dependencies:formatting): subject

commitDescription: subject

commitFixes: []

commitRefs: []

commitScopes: [dependencies, formatting]

commitType: chore

isType: false

revertedCommit: 

----------------------------
chore: subject (refs ABC-123)

commitDescription: subject

commitFixes: []

commitRefs: [ABC-123]

commitScopes: []

commitType: chore

isType: false

revertedCommit: 

----------------------------
chore: subject (refs ABC-123 DEF-456)

commitDescription: subject

commitFixes: []

commitRefs: [ABC-123, DEF-456]

commitScopes: []

commitType: chore

isType: false

revertedCommit: 

----------------------------
chore: subject (fixes ABC-123

commitDescription: subject

commitFixes: [ABC-123]

commitRefs: []

commitScopes: []

commitType: chore

isType: false

revertedCommit: 

----------------------------
Revert "doing change 1"

This reverts commit 114a9170d538c35b2fe49ba81453e915192e2d8e.

commitDescription: 

commitFixes: []

commitRefs: []

commitScopes: []

commitType: revert

isType: false

revertedCommit: 114a9170d538c35b2fe49ba81453e915192e2d8e

